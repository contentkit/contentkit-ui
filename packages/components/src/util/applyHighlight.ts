import { Modifier, EditorState } from 'draft-js'

function applyHighlight (editorState, selection) {
  const contentState = Modifier.applyInlineStyle(
    editorState.getCurrentContent(),
    selection,
    'HIGHLIGHT'
  )
  return EditorState.push(
    editorState,
    contentState,
    'change-inline-style'
  )
}

export default applyHighlight
