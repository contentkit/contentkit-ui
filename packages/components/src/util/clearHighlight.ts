
import { Modifier, EditorState, SelectionState } from 'draft-js'

function clearHighlight (editorState, selection) {
  const contentState = Modifier.removeInlineStyle(editorState.getCurrentContent(), selection, 'HIGHLIGHT')
  const nextEditorState = EditorState.push(editorState, contentState, 'change-inline-style')
  return nextEditorState
  // return EditorState.forceSelection(nextEditorState, selection)
}

export default clearHighlight