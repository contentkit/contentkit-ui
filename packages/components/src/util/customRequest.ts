import { CustomRequestParams } from '../types'

async function customRequest ({ headers, file, action, onSuccess }: CustomRequestParams): Promise<void> {
  const formData = new window.FormData()
  for (let field in action.fields) {
    formData.append(field, action.fields[field])
  }
  formData.append('file', file)
  const resp = await fetch(action.url, {
    method: 'POST',
    body: formData,
    headers: headers
  })
  onSuccess(resp)
}

export default customRequest
