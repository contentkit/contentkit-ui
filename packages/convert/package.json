{
  "name": "@contentkit/convert",
  "version": "0.0.62",
  "main": "lib/index.js",
  "private": false,
  "license": "MIT",
  "scripts": {
    "clean": "rimraf lib",
    "build": "yarn run clean && tsc -p ./config",
    "watch": "tsc-watch -p ./config",
    "lint": "standard \"src/**/*.js\" --fix --verbose | snazzy",
    "test": "jest"
  },
  "standard": {
    "parser": "babel-eslint"
  },
  "dependencies": {
    "@contentkit/types": "^0.0.62",
    "@contentkit/util": "^0.0.62",
    "draft-js-export-html": "^1.4.1",
    "html-element-attributes": "^2.2.1",
    "lodash.groupby": "^4.6.0",
    "lodash.pick": "^4.4.0",
    "prismjs": "^1.20.0"
  },
  "peerDependencies": {
    "draft-js": "0.11.6",
    "immutable": "3.7.6",
    "jest": "^24.5.0"
  },
  "publishConfig": {
    "registry": "https://registry.npmjs.org/",
    "access": "public"
  },
  "devDependencies": {
    "@types/immutable": "3.8.1",
    "draft-js": "^0.11.6",
    "immutable": "3.7.6",
    "tsc-watch": "3.0.2",
    "typescript": "^3.7.4"
  },
  "gitHead": "af83b1d1596f29fc61c474034b39297155903e59"
}
