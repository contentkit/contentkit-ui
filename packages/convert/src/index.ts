import * as blockConverters from './blockConverters'
import convertToHTML from './convertToHTML'
import contentStateToHTML from './contentStateToHTML'

export {
  convertToHTML,
  blockConverters,
  contentStateToHTML
}

export default convertToHTML

