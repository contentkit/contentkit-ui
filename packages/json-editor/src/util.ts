
import { OPEN_SCOPE_DELIMETERS, CHARACTER_PAIRS } from './fixtures'

export const pad = (offset) => new Array(offset).fill(' ').join('')

export const DEFAULT_INDENT = pad(2)


export function isCharacterPair (left: string, right: string) {
  return OPEN_SCOPE_DELIMETERS.includes(left) && right === CHARACTER_PAIRS[left]
}

