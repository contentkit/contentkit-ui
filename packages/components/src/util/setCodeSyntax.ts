import { EditorState, SelectionState, Modifier, EditorChangeType } from 'draft-js'
import { Map } from 'immutable'
import { getContinuousBlocks, ChangeType, createEntity, Entity, Syntax } from '@contentkit/util'

type Store = {
  getEditorState: () => EditorState,
  setEditorState: (editorState: EditorState) => void
}

function setCodeSyntax (value: string, selectionState, { getEditorState, setEditorState }: Store): void {
  const editorState = getEditorState()
  const nextEditorState = createEntity(editorState, selectionState, Entity.CODE, { [Syntax.KEY]: value })
  setEditorState(nextEditorState)
}

export default setCodeSyntax
