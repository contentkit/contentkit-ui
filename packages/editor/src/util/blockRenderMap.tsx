import React from 'react'
import { Map } from 'immutable'
import { Block } from '@contentkit/util'
import { DefaultDraftBlockRenderMap } from 'draft-js'

const CODE_WRAP = <pre className='cke-code-container' />
const OL_WRAP = <ol className='public-DraftStyleDefault-ul' />
const UL_WRAP = <ul className='public-DraftStyleDefault-ul' />

const customBlockRenderMap = Map({

  [Block.UNSTYLED]: {
    element: 'p'
  },
  [Block.BLOCKQUOTE]: {
    element: 'div',
    wrapper: <blockquote />
  },
  [Block.TODO]: {
    element: 'div'
  },
  [Block.IMAGE]: {
    element: 'figure'
  },
  [Block.VIDEO]: {
    element: 'figure'
  },
  [Block.CODE]: {
    element: 'div',
    wrapper: CODE_WRAP
  },
  [Block.OL]: {
    element: 'li',
    wrapper: OL_WRAP
  },
  [Block.UL]: {
    element: 'li',
    wrapper: UL_WRAP
  },
  [Block.TABLE]: {
    element: 'div'
  },
  [Block.GIST]: {
    element: 'div'
  },
  [Block.SCRIPT]: {
    element: 'div'
  }
})

export default DefaultDraftBlockRenderMap.merge(customBlockRenderMap)
