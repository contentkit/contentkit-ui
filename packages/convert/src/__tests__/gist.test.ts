import { blockConverters } from '@contentkit/convert'
import { EditorState, SelectionState, ContentState, ContentBlock, CharacterMetadata } from 'draft-js'
import { Block } from '@contentkit/util'
import { Map, List, Repeat } from 'immutable'

const getEditorState = (blockArray, focusKey) => {
  const editorState = EditorState.createWithContent(
    ContentState.createFromBlockArray(blockArray)
  )
  return EditorState.acceptSelection(editorState,
    new SelectionState({
      anchorKey: focusKey,
      anchorOffset: 1,
      focusKey: focusKey,
      focusOffset: 1,
      isBackward: false
    })
  )
}

it('convertGist', () => {
  const blocksArray = [
    new ContentBlock({ key: 'A' }),
    new ContentBlock({
      key: 'B',
      type: Block.GIST,
      text: ' ',
      characterList: List(Repeat(new CharacterMetadata(), 1)),
      data: Map({
        tag: '<script src="https://gist.github.com/unshift/c7cc5cce1fc2cf034adab9586c7acae8.js"></script>'
      })
    }),
    new ContentBlock({ key: 'D' })
  ]

  const editorState = getEditorState(blocksArray, 'C')
  const contentState = editorState.getCurrentContent()
  const contentBlock = contentState.getBlockForKey('B')
  expect(
    blockConverters.gist(contentState, contentBlock)
  ).toMatchSnapshot()
})
