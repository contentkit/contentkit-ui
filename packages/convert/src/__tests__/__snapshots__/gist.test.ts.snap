// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convertGist 1`] = `"<script src=\\"https://gist.github.com/unshift/c7cc5cce1fc2cf034adab9586c7acae8.js\\"></script>"`;
